tests/test_models/test_attachments.py

```diff
@@ -0,0 +1,170 @@
+import pytest
+
+from botx import MessageBuilder
+
+
+def test_is_link_in_attachment():
+    builder = MessageBuilder()
+    builder.link()
+    assert builder.message.attachments.__root__[0].data.is_link()
+
+
+def test_is_mail_in_attachment():
+    builder = MessageBuilder()
+    mailto_url = "mailto:mail@mail.com"
+    builder.link(url=mailto_url)
+    assert builder.message.attachments.__root__[0].data.is_mail()
+
+
+def test_is_telephone_number_in_attachment():
+    builder = MessageBuilder()
+    tel_url = "tel://+77777777777"
+    builder.link(url=tel_url)
+    assert builder.message.attachments.__root__[0].data.is_telephone()
+
+
+def test_mailto_property_in_attachment():
+    builder = MessageBuilder()
+    mailto_url = "mailto:mail@mail.com"
+    builder.link(mailto_url)
+    assert builder.message.attachments.__root__[0].data.mailto == "mail@mail.com"
+
+
+def test_raising_missing_mailto():
+    builder = MessageBuilder()
+    builder.link()
+    with pytest.raises(AttributeError):
+        builder.message.attachments.__root__[0].data.mailto
+
+
+def test_tel_property_in_attachment():
+    builder = MessageBuilder()
+    tel_url = "tel://+77777777777"
+    builder.link(url=tel_url)
+    assert builder.message.attachments.__root__[0].data.tel == "+77777777777"
+
+
+def test_raising_missing_tel():
+    builder = MessageBuilder()
+    builder.link()
+    with pytest.raises(AttributeError):
+        builder.message.attachments.__root__[0].data.tel
+
+
+def test_image_in_attachments():
+    builder = MessageBuilder()
+    builder.document()
```

Comment:
Why theese test uses both methods, not one?

Reply:
because this method iterated over **list** of attachments.

Reply:
List from one element is also a list. Or I don't understand your thought.